### PR TITLE
count inversion.cpp

### PR DESCRIPTION
**Pull request by mansikagrawal**
#1664 
Fixes #

#### Checklist:
- [Done ] 4 space indentation.
- [ Done] Coding conventions are followed.
- [ Done] Input is taken dynamically.
- [ Done] Sample Input / Output is added at the end of file.
- [ Done] Logic Documentation (Comments).
- [ Done] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request: Added code count inversion.cpp
- 

#### Languages Used: cpp
- 
- 

#### Files Added: count inversion.cpp
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Count_Inversion/count inversion.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#include<iostream>
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: please give two line description about count inversion at the top

File: Count_Inversion/count inversion.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+using namespace std;
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+int mergefn(int arr[],int temp[],int left,int mid,int right)
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space after comma

File: Count_Inversion/count inversion.cpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+{
+// temp[] is the auxillary array to store the sorted elements
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: move this commend before the mergefn() function

File: Count_Inversion/count inversion.cpp [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove these lines

File: Count_Inversion/count inversion.cpp [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    int i=left;
+    int j=mid;
+    int k=left;
+    int inv_cnt=0;
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after comma

File: Count_Inversion/count inversion.cpp [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove these two lines

File: Count_Inversion/count inversion.cpp [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+    while((i<=mid-1 ) && (j<=right))
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator

File: Count_Inversion/count inversion.cpp [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    {
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+        if(arr[i]<=arr[j])
+            temp[k++]=arr[i++];
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator

File: Count_Inversion/count inversion.cpp [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        else
+        {
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+            temp[k++]=arr[j++];
+            inv_cnt+=(mid-i);
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator

File: Count_Inversion/count inversion.cpp [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+            
+            /* 
+             if arr[i] is greater than arr[j], then there are (mid – i) inversions. because left and right subarrays are sorted,
+             so all the remaining elements in left-subarray (arr[i+1], arr[i+2] … arr[mid]) will be greater than arr[j]*/
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: one line should only contain 80 characters. break the comment accordingly

File: Count_Inversion/count inversion.cpp [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove the lines

File: Count_Inversion/count inversion.cpp [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+
+    while(i<=mid-1)
+        temp[k++]=arr[i++];
+    while(j<=right)
+        temp[k++]=arr[j++];
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator

File: Count_Inversion/count inversion.cpp [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+
+
+
+    for(i=left;i<=right;i++)
+        arr[i]=temp[i];
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator

File: Count_Inversion/count inversion.cpp [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+
+    if(right>left)
+    {
+
+        int mid=(left+right)/2;
+
+
+        inv_cnt+=mergesort(arr,temp,left,mid);// counting inversions in left subarray
+        inv_cnt+=mergesort(arr,temp,mid+1,right);//counting inversions in the right subarray 
+        inv_cnt+=mergefn(arr,temp,left,mid+1,right);// count overall inversions while merging the left subarray and right subarray
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator
correct the indentation
keep space after comma 
take comments to previous line

File: Count_Inversion/count inversion.cpp [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+
+    }
+    return inv_cnt;
+}
+
+
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove the lines

File: Count_Inversion/count inversion.cpp [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+    int n;
+    cin>>n;
+    int arr[n];
+    for(int i=0;i<n;i++)
+        cin>>arr[i];
+
+    int temp[n];
+    cout<<mergesort(arr,temp,0,n-1)<<endl;
+    }
+    return 0;
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space before and after operator
correct the indentation
keep space after comma

File: Count_Inversion/count inversion.cpp [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+}
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 23]
```diff
@@ -0,0 +1,106 @@
...
+        temp[k++]=arr[j++];
+
+
+
+    for(i=left;i<=right;i++)
+        arr[i]=temp[i];
+
+    return inv_cnt;
+
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove these two line

File: Count_Inversion/count inversion.cpp [hunk 24]
```diff
@@ -0,0 +1,106 @@
...
+}
+
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: remove this line

File: Count_Inversion/count inversion.cpp [hunk 25]
```diff
@@ -0,0 +1,106 @@
...
+
+int mergesort(int arr[],int temp[],int left,int right)
```
Comment (asha15) [inline Count_Inversion/count inversion.cpp]: keep space after comma

**Comment by jainaman224** [comment]
@mansikagrawal Are you working on it?

**Comment by mansikagrawal** [comment]
@jainaman224 yes I am

File: Count_Inversion/count inversion.cpp (hunk 2 above)
Comment (mansikagrawal) [reply Count_Inversion/count inversion.cpp]: Mam I couldn't get which line?

**Comment by mansikagrawal** [comment]
@asha15  I had made one more pull request regarding this issue please check #2242

File: Count_Inversion/count inversion.cpp (hunk 2 above)
Comment (asha15) [reply Count_Inversion/count inversion.cpp]: Third line

**Comment by somya-kapoor** [comment]
@mansikagrawal Please close the existing PR if you are creating a new PR regarding the same issue.

**Pull request closed without merging**